core-httpclient-impl/src/main/java/com/optimizely/ab/odp/DefaultODPApiManager.java

```diff
@@ -167,6 +182,43 @@ public String fetchQualifiedSegments(String apiKey, String apiEndpoint, String u
         return null;
     }
 
+    @Override
+    public void sendEvents(String apiKey, String apiEndpoint, List<ODPEvent> events) {
+        HttpPost request = new HttpPost(apiEndpoint);
+        String requestPayload = this.jsonSerializer.serializeEvents(events);
+
+        if (requestPayload == null || requestPayload.isEmpty()) {
+            logger.error("ODP event send failed (Failed to serialize event payload)");
+            return;
+        }
+
+        try {
+            request.setEntity(new StringEntity(requestPayload));
+        } catch (UnsupportedEncodingException e) {
+            logger.error("ODP event send failed (Error encoding request payload)", e);
+            return;
+        }
+        request.setHeader("x-api-key", apiKey);
+        request.setHeader("content-type", "application/json");
+
+        CloseableHttpResponse response = null;
+        try {
+            response = httpClient.execute(request);
+        } catch (IOException e) {
+            logger.error("Error retrieving response from event request", e);
+            return;
+        }
+
+        if (response.getStatusLine().getStatusCode() >= 400) {
```

Comment:
Failures with recoverable errors (status=5xx, connection errors, etc) should be differentiated so that OdpEventManager can retry them later.

Reply:
Now returning status code from the method so that consuming code can use it and retry when needed.